src/wms/src/test/java/org/geoserver/wms/legendgraphic/BufferedImageLegendGraphicOutputFormatTest.java

```diff
@@ -252,15 +260,58 @@ public void testMultipleLayers() throws Exception {
 
         assertPixel(image, 10, 50 + titleHeight, new Color(224, 64, 0));
 
-        // second title
-        assertPixel(image, 1, 60 + titleHeight + titleHeight / 2, new Color(0, 0, 0));
-
         // same colors for the second layer
         assertPixel(image, 10, 70 + titleHeight * 2, new Color(192, 160, 0));
 
         assertPixel(image, 10, 90 + titleHeight * 2, new Color(0, 0, 0));
 
         assertPixel(image, 10, 110 + titleHeight * 2, new Color(224, 64, 0));
+
+        // 128 * 3 = search for pixels are at least half black  (3 = number of components - RGB)
+        assertManyPixelsOfColor(image, Color.BLACK, 128 * 3, true, 1500);
+    }
+
+    /**
+     * Looks through the image and counts the number of pixels that are approximately the given
+     * color. This must be greater than minNumber.
+     *
+     * @param image image to look through
+     * @param color test color
+     * @param tolerance how close the pixel's test color needs to be to be a match (0=exact match)
+     * @paaram onlyGrey only consider grey-scale pixels
```

Comment:
This did not quite do what I expected based on the method signature.

Really you are only testing pixels where all the components are the same (ie gray), and skipping testing anything else. I had assumed you were looking at the ``HSV`` of the comments and only going to check the image ``V`` component (ie a grayscale comparison of everything).

Consider:
``
onlyGray Filter all colour pixels leaving only gray pixels to be compared.
``

Reply:
changed